openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -220,15 +232,25 @@ def check_heketi_db_inconsistencies(
             error_msg)
 
     def create_secret(self, secret_name_prefix="autotests-secret",
-                      secret_type=None, skip_cleanup=False):
-        secret_name = oc_create_secret(
-            self.ocp_client[0],
-            secret_name_prefix=secret_name_prefix,
-            namespace=(self.sc.get(
-                'secretnamespace',
-                self.sc.get('restsecretnamespace', 'default'))),
-            data_key=self.heketi_cli_key,
-            secret_type=secret_type or self.secret_type)
+                      secret_type=None, skip_cleanup=False, glusterfs_registry=False):
+        if glusterfs_registry:
+            secret_name = oc_create_secret(
+                self.ocp_client[0],
+                secret_name_prefix=secret_name_prefix,
+                namespace=(self.registry_sc.get(
+                    'secretnamespace',
+                    self.registry_sc.get('restsecretnamespace', 'default'))),
+                data_key=self.registry_heketi_cli_key,
+                secret_type=secret_type or self.registry_secret_type)
+        else:
+            secret_name = oc_create_secret(
+                self.ocp_client[0],
+                secret_name_prefix=secret_name_prefix,
+                namespace=(self.sc.get(
+                    'secretnamespace',
+                    self.sc.get('restsecretnamespace', 'default'))),
+                data_key=self.heketi_cli_key,
+                secret_type=secret_type or self.secret_type)
```

Comment:
Instead of this,
I would suggest having a different variable for sc and use that.
```
       if glusterfs_registry:
            self.sc = self.registry_sc.get('restsecretnamespace', 'default')
        else:
            self.sc = self.sc.get('restsecretnamespace', 'default') 
        secret_name = oc_create_secret(
            self.ocp_client[0],
            secret_name_prefix=secret_name_prefix,
            namespace=(self.sc.get('secretnamespace',self.sc)),
            data_key=self.heketi_cli_key,
            secret_type=secret_type or self.secret_type)
```

Reply:
@SushilG96 Here at a time only one sc and secret will get created. and you cannot create multiple sc from both the projects at once.

Reply:
Yes, its okay